internal/imodels/imodels.go

```diff
@@ -32,123 +32,140 @@ var osExtractors = map[string]struct{}{
 	rpm.Extractor{}.Name():  {},
 }
 
-// PackageInfo represents a package found during a scan. This is generally
-// converted directly from the extractor.Inventory type, with some restructuring
-// for easier use within osv-scanner itself.
+// PackageInfo provides getter functions for commonly used fields of inventory
+// and applies transformations when required for use in osv-scanner
 type PackageInfo struct {
-	Name      string // Name will be SourceName matching the osv-schema
-	Version   string
-	Ecosystem ecosystem.Parsed
-
-	Location   string // Contains Inventory.Locations[0]
-	SourceType SourceType
-
-	Commit     string
-	Repository string
+	// purlCache is used to cache the special case for SBOMs where we convert Name, Version, and Ecosystem from purls
+	// extracted from the SBOM
+	purlCache *models.PackageInfo
+	*extractor.Inventory
+}
 
-	// For package sources
-	DepGroups []string
+func (pkg *PackageInfo) Name() string {
+	// TODO(v2): SBOM special case, to be removed after PURL to ESI conversion within each extractor is complete
+	if pkg.purlCache != nil {
+		return pkg.purlCache.Name
+	}
 
-	// For OS packages
-	// This is usually the BinaryName, while Name is the SourceName
-	OSPackageName string
+	if pkg.Ecosystem().Ecosystem == osvschema.EcosystemGo && pkg.Inventory.Name == "go" {
+		return "stdlib"
+	}
 
-	AdditionalLocations []string // Contains Inventory.Locations[1..]
+	if metadata, ok := pkg.Inventory.Metadata.(*dpkg.Metadata); ok {
+		// Debian uses source name on osv.dev
```

Comment:
Do we need to add a TODO for any other distros?

Reply:
I don't think so, at least not for any of the distros we currently support (redhat, alpine), SourceName is not in either of those metadatas.